src/UIComponents/Predict.jsx

```diff
@@ -30,7 +30,7 @@ class Predict extends Component {
   };
 
   handleChange = (event, feature) => {
-    const testData = this.props.testData;
+    const testData = {...this.props.testData};
```

Comment:
Ah, without this, React doesn't think a prop has changed, and doesn't re-render.